example/dj/apps/issue_tracker/tests/rest_permissions.py

```diff
@@ -1,147 +1,93 @@
+        assert_http_bad_request(self.put('/api/issue/{}/'.format(issue.pk), {}))
+        assert_http_accepted(self.delete('/api/user/{}/'.format(user.pk)))
+        assert_http_accepted(self.delete('/api/issue/{}/'.format(issue.pk)))
 
     @login(is_superuser=False)
-    def test_user_can_read_only_itself(self):
-        resp = self.get(('%s%s/') % (self.USER_API_URL, self.logged_user.user.pk))
-        assert_valid_JSON_response(resp)
-
-        user = self.get_user_obj()
-        resp = self.get(('%s%s/') % (self.USER_API_URL, user.pk))
-        assert_http_forbidden(resp)
-
-    @login(is_superuser=True)
-    def test_superuser_can_add_new_user(self):
-        resp = self.post(self.USER_API_URL, data=self.get_user_data())
-        assert_valid_JSON_created_response(resp)
+    def test_user_without_permission_should_do_nothing(self):
+        issue = self.create_issue()
+        user = self.create_user('new_user', 'password', 'test@email.com')
+
+        # API
+        # Generic read, post
+        assert_http_forbidden(self.get('/api/user/'))
+        assert_http_forbidden(self.get('/api/issue/'))
+        assert_http_forbidden(self.post('/api/user/', {}))
+        assert_http_forbidden(self.post('/api/issue/', {}))
+
+        # API
+        # Generic read, put, patch, delete
+        assert_http_forbidden(self.get('/api/user/{}/'.format(user.pk)))
+        assert_http_forbidden(self.get('/api/issue/{}/'.format(issue.pk)))
+        assert_http_forbidden(self.put('/api/user/{}/'.format(user.pk), {}))
+        assert_http_forbidden(self.put('/api/issue/{}/'.format(issue.pk), {}))
+        assert_http_forbidden(self.delete('/api/user/{}/'.format(user.pk)))
+        assert_http_forbidden(self.delete('/api/issue/{}/'.format(issue.pk)))
 
     @login(is_superuser=False)
-    def test_only_superuser_can_add_new_user(self):
-        resp = self.post(self.USER_API_URL, data=self.get_user_data())
-        assert_http_forbidden(resp)
+    def test_user_with_permission_should_do_allowed_operations(self):
+        self.sync_permissions()
+        issue = self.create_issue()
+        user = self.create_user('new_user', 'password', 'test@email.com')
 
-    @login(is_superuser=True)
-    def test_superuser_can_add_update_user(self):
-        user = self.get_user_obj()
-        resp = self.put('%s%s/' % (self.USER_API_URL, user.pk), data={})
-        assert_valid_JSON_response(resp)
-
-    @login(is_superuser=False)
-    def test_user_can_update_only_itself(self):
-        user = self.get_user_obj()
-        resp = self.put('%s%s/' % (self.USER_API_URL, user.pk), data={})
-        assert_http_forbidden(resp)
+        logged_user = self.logged_user.user
 
-        user = self.logged_user.user
-        resp = self.put('%s%s/' % (self.USER_API_URL, user.pk), data={})
-        assert_valid_JSON_response(resp)
+        issue_read_permission = Perm.objects.get(codename='{}__{}'.format('issue', 'read'))
```

Comment:
nebyl by na tohle fajn nějakej helper? se to tu opakuje

Reply:
jen 4x, je to v testech, to bych neřešil, zhoršilo by to čitelnost